Sound balance

FrequentPilgrim opened the pull request:
The next round of sound balancing.

GoogleFrog commented:
I've gone through and here are the things I picked up for sounding off. When I say "flat", I mean like the sound has been compressed and detail stripped away. The way the world sounds with blocked ears. I don't mean musically flat.
- Energy pylon selection sound is a little flat.
- Convict selection sound is a little flat.
- Bandit firing sound is a bit too weighty.
- Ronin firing sound is slightly too flat and has some weird swooshing at the end.
- Knight has some weird swooshing at the end.
- Phantom impact sound is hard to describe. It feels flat but also has too many details.
- Imp explosion sounds too flat.
- Dart has a little swoosh at the end and feels too impactful.
- Ravager impact might be a bit too flat.
- Dominatrix sounds too accentuated.
- Badger firing has a bit too much echo.
- Crasher impact has too much echo.
- Thunderbird weapon might be too flat and have too much echo. Hard to tell.
- Odin shield impact is off in a hard to describe way. This probably extends to Badger spam.
- Gnat weapon is too flat.
- Harpy impact has too much echo.
- Nimbus sounds more grating than the previous version. Perhaps it is a bit flatter so a repetitive part of the sound is shining through.
- Krow laser is too flat.
- Bolas sounds too weighty.
- Scalpel firing has a swoosh at the end.
- Halberd is a bit flat.
- Lance sounds really weird. Flatness and other effects. It doesn't sound like it is synchronised to the visuals the way the previous sound is. Possibly the previous sound ramps up a bit until the end then drops off.
- Archer impact has turned into an annoying echo.
- Grizzly weapon is very slightly too flat.
- Limpet detonation has more of a swooshy echo at the end.
- Hermit impact has a noticeable echo at the end.
- Widow is not impactful enough and sounds flat.
- Constable weapon sounds too impactful and accentuated.
- Firewalker impact point blank into a structure is just broken.
- Cyclops slowbeam sounds too impactful as well.
- Tremor firing is too flat.
- Siren sonic impact has the Archer problem.
- Siren missile impact has a lot of echo.
- Dante heatray is far too loud compared to its other weapons.
- Paladin laser and lighting are too flat.
- Det gauss is too quite for its other weapons.
- Stinger weapon is too flat.
- Lucifer has the Lance problem.
- Desolator heatray is comparatively loud.
- Hacksaw impacts have swoosh at the end.
- Chainsaw firing has a swoosh at the end.
- Tacnuke launches are too quiet.
- Starlight beam burst is too quiet and flat.
- DRP yellow shot sounds truncated.

I didn't test all of sea.

GoogleFrog commented:
Sorry for the delay, I've gone through it all again as a bit of a blind test.
- Phantom firing sound is a lot softer than its impact sound. This is a bit weird as they tend to make up the same action, so ends up sounding like a volume shift. The sounds could be closer in volume, with the firing going up more than the impact goes down.
- Imp explosion is too flat.
- Badger fire has an echoy whoosh at the end that it could do without.
- Harpy weapon impact is a bit too quiet compared to its firing sound.
- Dagger weaponry is too quite in general (fire and impact). At least, it is hard to hear over the sound of moving a Dagger around. Other raiders have louder weapon sounds.
- Is Lance firing still a bit too short? I'm not sure I can tell at this point. My feeling watching it is that the main part of the sound (not the aftermath) cuts out a fraction of a second before the beam does.
- Archer impact has an undesirable echoy part at the end.
- Widow shot is also a bit too flat, like Imp. Having this weapon announce itself to the entire battle is fine.
- Placeholder firing is really weird and the impact may be a bit quiet. It used to have more of a thoomp to it when it fired. This weapon announcing itself is also not so bad.
- Each individual Dart shot is louder than one Moderator shot. At least that is how it sounds when both are firing at once. Whether it is illusory or not, something should be changed to swap their loudnesses, since the Moderator shot is a lot bigger.
- Jack is slightly too flat and has this whooshy deep rumble at the end, that I don't recall being in the stable version.
- Point blank Firewalker is still broken. Spawn a Firewalker like this and hear for yourself. When fired at range, the shots are each individually too crisp, which breaks up the impact into a series of hits rather than the general laying of a carpet of fire from stable.
<img width="760" height="476" alt="image" src="https://github.com/user-attachments/assets/7f6efdad-2ed1-4afd-bd38-bcdcd8d64992" />

- Siren sonic has the same whooshy echo at the end as Archer.
- Paladin lightning is much too flat, and much quieter than its laser.
- Detriment gauss should be louder to not be drowned out by the laser. I think this much gauss being a bit of an assault on the senses is fine, it's Detriment.
- Shogun may be a bit too quiet, especially its impacts.
- Reef missile firing is too quiet.
- Lotus has a slight repetitive whir when it fires. Something apart from the main hiss of the weapon.
- Gauss firing is loud enough, but there is something weird about the impact. Perhaps each impact has too much punch at the start, making multiple impacts sound a bit repetitive. Idk.
<img width="982" height="652" alt="image" src="https://github.com/user-attachments/assets/0a554ced-c192-4414-b2f6-466291ab0cc9" />

- Lucifier sounds wrong, compared to Lance, and I don't know why. Fixing Lance probably solves it.
- Thresher (and Ettin) may be a bit too loud/obvious compared to similar units. If you want to make things fade into each other, there is probably room here. Also there a whoosh at the end of the flak firing sound.
- The five missile silo missiles should have louder launch sounds. The Scylla launch sound has an appropriate volume.
- The Zenith meteor impact sound is too crisp for when a cluster of them hit the ground. The individual impacts by themselves may also be too flat.

Upon reviewing the two lists I see a lot of progress. The plasma cannon issues seems generally resolved, and some swooshes are gone. There are still a few swooshes and some mismatched weapon/impact volumes (imo).
- On firing and impact volumes, my theory is usually (depends on the projectile) that the firing and hitting are the same auditory action. So it feels like one side of the sound is missing when one is a lot louder than the other.
- On crisp barrages, my goal with something like Firewalker or Zenith is for the barrage to sound more continual than it does at the moment. Hearing individual impacts (to the extent that I can in this version) sounds too artificial/gamey.
- Forking sounds to solve the above (or other remaining issues) is fine. Eg perhaps Paladin and Widow will want to end up with different lightning sounds, based on the same file, but varied someone to reflect their different importance.

GoogleFrog commented:
Here is a deep dive on Firewalker and the crispness of a barrage. I've attached three sets of Firewalker shots and impacts, first with the old version and then with the new.
[OldNewFirewalker2025-08-04 11-19-45.zip](https://github.com/user-attachments/files/21569225/OldNewFirewalker2025-08-04.11-19-45.zip)

The old Firewalker impact sound is muddier, but this works in its favour as it lets the lingering muddiness of the early impact sounds mess with the impacts of the later sounds. Look at the old vs new impact, you can see the individual peaks. Here is a picture of the third set of trials in my attachment.

<img width="1288" height="346" alt="image" src="https://github.com/user-attachments/assets/8a56d9d9-171a-4fe6-acce-3daa51ace6ec" />

The one I've pointed to is clearly audible as an artificial pop. We shouldn't be able to see (or hear) the peaks, as they remind the player that they are hearing a simulation of some small number of projectiles at a fixed frame rate. The fiction of Firewalker is that it is laying a carpet of fire, so we shouldn't be hearing individual impacts so clearly.

GoogleFrog commented:
Here is what I think is going on with Lance. The waveform isn't that helpful, but it gives me something to refer to. Here is old and new Lance, as heard ingame:
<img width="832" height="721" alt="image" src="https://github.com/user-attachments/assets/235fa4cf-ab1f-488f-888c-05e45a4b5c82" />

Just eyeballing the waveforms, old Lance sustains for around 1.5s while new starts falling off 0.5s in. Audibly there are a few more distinct sounds involved with the old Lance sound.
- The first 0.6 seconds make up the punch of the initial weapon firing. This is the part that the new sound accentuates.
- From around 0.55s to 0.9s, the sound of the initial fire is bridged by a region of extreme clipping that sounds a bit like VwaVwaVa when you zoom into it.
- At around 1s the "debris flying sound" swells and sustains at that intensity until around 1.3s. This gives the end of the beam its own audio cue.

The new sound lacks the secondary swell of the above, leaving the end feeling lackluster.

FrequentPilgrim commented:
- Phantom firing sound is a lot softer than its impact sound. This is a bit weird as they tend to make up the same action, so ends up sounding like a volume shift. The sounds could be closer in volume, with the firing going up more than the impact goes down.
**_Phantom has been bugging me for a long time. It's too unique to be sharing sounds with another unit, which also frees up detriment and lance to be more specialized. I did an overhaul on both the firing and hit. Overlaid some unique texture and frequency that sells the fantasy of the invisible sniper. I pushed it fairly hard with the same philosophy of the glaive - Loud, Clear, Unique._** 
- Imp explosion is too flat.
**_Imp is restored in frequency._**
- Badger fire has an echoy whoosh at the end that it could do without.
**_Cut tail off the badger fire sound further._**
- Harpy weapon impact is a bit too quiet compared to its firing sound.
**_Boosted harpy impact_**
- Dagger weaponry is too quite in general (fire and impact). At least, it is hard to hear over the sound of moving a Dagger around. Other raiders have louder weapon sounds.
**_see below_**
- Gauss firing is loud enough, but there is something weird about the impact. Perhaps each impact has too much punch at the start, making multiple impacts sound a bit repetitive. Idk.
**_Softened the attack on gauss_small. Raised volume on dagger and slightly boosted gain on both sounds to better match. Hearing the quick succession of hits on piercing dmg was a bit jarring but the softened attack helps._**
- Is Lance firing still a bit too short? I'm not sure I can tell at this point. My feeling watching it is that the main part of the sound (not the aftermath) cuts out a fraction of a second before the beam does.
**_See Below_**
- Lucifier sounds wrong, compared to Lance, and I don't know why. Fixing Lance probably solves it.
**_Recreated the Lance sound to have lots of presence and no falloff until after the beam has fully ended._**
- Archer impact has an undesirable echoy part at the end.
**_See Below_**
- Siren sonic has the same whooshy echo at the end as Archer.
**_Further removed tail from archer and siren_**
- Widow shot is also a bit too flat, like Imp. Having this weapon announce itself to the entire battle is fine.
**_Gave it a lot of extra presence_**
- Placeholder firing is really weird and the impact may be a bit quiet. It used to have more of a thoomp to it when it fired. This weapon announcing itself is also not so bad.
**_Boosted the placeholder as much as possible using original source._**
- Each individual Dart shot is louder than one Moderator shot. At least that is how it sounds when both are firing at once. Whether it is illusory or not, something should be changed to swap their loudnesses, since the Moderator shot is a lot bigger.
**_Boosted moderator. Dart seemed good volume wise_**
- Jack is slightly too flat and has this whooshy deep rumble at the end, that I don't recall being in the stable version.
**_Removed tail from jack_**
- Point blank Firewalker is still broken. Spawn a Firewalker like this and hear for yourself. When fired at range, the shots are each individually too crisp, which breaks up the impact into a series of hits rather than the general laying of a carpet of fire from stable.
**_Default Firewalker is fundamentally broken in a way that is hard to address. The pleasing mud that covers the impacts you're describing are the result of sub 40hz frequencies that make up a huge amount of the firewalker impact. 99% of Sound systems can't produce those frequencies, and instead produce distortion. This distortion is the mud that's working as glue. The problem is that every sound system is going to handle that distortion and the sounds on top of it differently, and before the sound rework there was so much distortion all the time that it flew under the radar. Using intentional distortion as an effect reliant on the limitations of each persons sound system while hurting the overall soundscape is probably not the route we want to go. There's no good way to quality control a sound like that and everyone will hear it differently. I've found a middle ground in taking the original and only doing a light cut of frequencies below 40hz and a slightly softened attack._** 


- Paladin lightning is much too flat, and much quieter than its laser.
**_Boosted presence and volume for paladin lightning. Paladin lightning may be too loud now but it's an easy adjustment downward._**
- Detriment gauss should be louder to not be drowned out by the laser. I think this much gauss being a bit of an assault on the senses is fine, it's Detriment.
**_Gave detriment even more presence and volume. Master -0db 40db highpass_**
- Shogun may be a bit too quiet, especially its impacts.
**_Boosted shogun impact_**
- Reef missile firing is too quiet.
**_Boosted missile firing launch sounds_** 
- Lotus has a slight repetitive whir when it fires. Something apart from the main hiss of the weapon.
**_Rebalanced the frequency to minimize the whirring. Original has this too, but it's better incorporated now._** 
- Thresher (and Ettin) may be a bit too loud/obvious compared to similar units. If you want to make things fade into each other, there is probably room here. Also there a whoosh at the end of the flak firing sound.
**_Shortened and adjusted the ettin and flak turret fire sounds to have a less prominent tonal sound. Can still reduce volume if necessary._** 

- The five missile silo missiles should have louder launch sounds. The Scylla launch sound has an appropriate volume.
**_Boosted missile firing launch sounds_** 
- The Zenith meteor impact sound is too crisp for when a cluster of them hit the ground. The individual impacts by themselves may also be too flat.
**_Split zenith off from cyclops. Gave it more presence. I'm liking the impact of it but I can soften the attack if needed. This is another one that was using unreliable distortion to smooth out impacts like the firewalker. I again tried to find a happy middleground._**

Upon reviewing the two lists I see a lot of progress. The plasma cannon issues seems generally resolved, and some swooshes are gone. There are still a few swooshes and some mismatched weapon/impact volumes (imo).
- On firing and impact volumes, my theory is usually (depends on the projectile) that the firing and hitting are the same auditory action. So it feels like one side of the sound is missing when one is a lot louder than the other.
- On crisp barrages, my goal with something like Firewalker or Zenith is for the barrage to sound more continual than it does at the moment. Hearing individual impacts (to the extent that I can in this version) sounds too artificial/gamey.
- Forking sounds to solve the above (or other remaining issues) is fine. Eg perhaps Paladin and Widow will want to end up with different lightning sounds, based on the same file, but varied someone to reflect their different importance.

**_Gave stinger a cleaner frequency curve and a longer duration before fadeout_**

_**Volume adjustments and tweaks are going to be an ongoing thing but it's very easy/quick to do at this point.  I'm most concerned with getting the foundation of the sound right since anyone in the future can adjust volumes on the files. Also, once we have live players in live matches I think the slight volume tweaks will be easier to spot and rapidly implement. But it would take some doing to get someone new up to speed on doing reworks and frequency profiles. **_

GoogleFrog commented:
I just found out that I had a problem with updating the changes, so I've rechecked and gone through everything. Sorry for anything extra that is here. Here are some comments as of https://github.com/ZeroK-RTS/Zero-K/pull/5520/commits/b1152edf0425838c25eec8f5b806ce63c514d8be, since there were a few reworks since.
- Bandit doesn't sound like it does much damage. The high pitched tap when it fires does not covey its level of power.
- Scythe seems to have an extra sound added on the slash. It sounds weird to me, like two things are happening in one action.
- The cleaned up old Phantom is fine by me. The new one feels like it doesn't fit. Perhaps there is something that could be done with the impact sound.
- Gremlin doesn't sound like it deals damage.
- Scorcher sounds very weird/truncated. It also doesn't really sound like it has much damage.
- Ripper impact sound seems very low-fidelity compared to the rest. Jarring.
- Impaler shouldn't have the same impact sound as Revenant. They're different types of rocket. The old sound was meant to convey the lack of AoE.
- Raptor also doesn't sound like it deals damage.
- I don't think Nimbus having essentially the same sounding gun as Reaver works.
- Hercules AA has the same problem as Raptor.
- This version of Dagger has a bit of a whistle in its impact, and lacks weight on firing?
- I don't get Halberd weapon impact. It sounds like a small building explosion rather than some sort of sci-fi magic missile.
- Latest Lance is an improvement on https://github.com/ZeroK-RTS/Zero-K/pull/5520/commits/b1152edf0425838c25eec8f5b806ce63c514d8be
- Buoy has the magic-energy firing sound without a matching impact, same as Halberd.
- Widow is fine here.
- Moderator firing sound might be too short. It doesn't sound like it matches its impact.
- Placeholder is fine here too.
- This Jack is good.
- Same Firewalker problem
- Toad doesn't sound like it deals damage.
- Ogre impact is offputting. The sound is very crackly.
- I think this is a better Tremor firing sound, but I can't really hear the impacts.
- Ettin sounds quite flat.
- The heatrays in Dante sound a bit unusual like Scorcher.
- The latest Paladin is good.
- This Detriment gauss is good. AA doesn't sound like it deals damage.

I don't have a lot of time at the moment so I haven't been as detailed as I would like. The changes up to the present (https://github.com/ZeroK-RTS/Zero-K/pull/5520/commits/aac1dd77c90336d4e2854fb64f1b98f0193383b2) are generally improvements. Here is what I would change:
- Keep the previous Phantom sound. Or even just keep the firing sound. I tested  `soundStart = weapon/gauss_fire` with the previous volume of the last commit and it works pretty well with the new impact sound. The new impact could be a bit brighter though, more dramatic.
- The sustained windy part of the impact sound for Placeholder is quite loud, it should be quieter.
- Revert the Jack sound changes in https://github.com/ZeroK-RTS/Zero-K/pull/5520/commits/c911dc39bc2ebf7333a5100b8898ec3e277dae1c

GoogleFrog commented:
I am also not sure how I feel about sounds like Tremor and the new Zeno sound. They may end up working, but for now they sound a bit toy-like.

FrequentPilgrim commented:
I'm going to give these all another look. You'll notice there's a through-line  for the sounds that are the most difficult, so I need to set expectations given the reality of the situation. **Rapid firing units: scorcher, bandit, raptor, impacts of zeno, tremor, nimbus, etc.** . If these rapid sounds are balanced, they lack impact. But if the impact is restored, they destroy the soundscape by smothering every other sound under them. I don't believe there is an actual "sweet spot" and instead these goals are mutually exclusive where there has to be a compromise since having impact is a lesser priority than the overall clarity of the soundscape. I say this because I'm predicting if I go in and start increasing all of the power and presence of the rapid firing units to the point where test firing them gives the feeling of "this sounds powerful" in the context of the test environment, it's going to be an overall regression. That isn't to say they can't be improved from where they are now, I want to push up to that line, and I'm going to give them all another pass. But going forward it's important to keep in mind when listening to any rapid-firing unit that this is a fundamentally compromised aspect of the game and we've already landed on the least-bad compromise of preserving every other unit's clarity at the expense of rapid firing unit's having less presence. If a bandit or scorcher or raptor etc. is preventing a player from easily following the action happening in their lane or makes them reach to turn down the volume slider, that's a far greater sin than those units being perfectly legible but lacking power.

FrequentPilgrim commented:
https://youtu.be/NZNj30qJtKs

Here's a video that demonstrates what I'm trying to communicate about rapid firing units. This example is bandits but it applies to virtually all of the rapid firing units. 12 bandits costing 900 metal are able to dominate the soundscape against a collection of more costly and more powerful units. My feeling is that the bandit is probably already a step too prominent, as it's not an exceptional case to field 12 of them or more. Anything to make a single bandit sound more powerful (more bass, more volume, longer duration, sharper attack) is going to have negative cascading effects when it gets multiplied.  I believe the bandit fires 10 shots per second, so with 12 we're triggering 120 audio cues every second. I've cut down the bandits audio file by more than half to 0.23 seconds, which means 2 more audio triggers occur before the first audio trigger has finished. That means for 12 bandits firing there are over 250 simultaneous audio clips playing constantly layered over each other.

FrequentPilgrim commented:
• Bandit doesn't sound like it does much damage. The high pitched tap when it fires does not covey its level of power.

-     • Restored some weight to the bandit.

    • Scythe seems to have an extra sound added on the slash. It sounds weird to me, like two things are happening in one action.

-     • Instead of blending the old slash with the new, now it's just the higher quality new slash. This may be a regression, it sounds better but is less piercing.

    • The cleaned up old Phantom is fine by me. The new one feels like it doesn't fit. Perhaps there is something that could be done with the impact sound.

-     • returned it back to gauss. May be too loud now, but easy to adjust back down to 7.5

    • Gremlin doesn't sound like it deals damage.

- Reworked Teal AA laser

    • Scorcher sounds very weird/truncated. It also doesn't really sound like it has much damage.

-   Reworked scorcher

    • Ripper impact sound seems very low-fidelity compared to the rest. Jarring

-     • Layered in a new sound underneath. 8000hz sample rate was bad.

    • Impaler shouldn't have the same impact sound as Revenant. They're different types of rocket. The old sound was meant to convey the lack of AoE.

-     • Created a new sound for impaler 

    • Raptor also doesn't sound like it deals damage.

- Reworked Teal AA laser

    • I don't think Nimbus having essentially the same sounding gun as Reaver works.

- Tried reworking Nimbus. I've tried about 10 different reworks of the nimbus and I'm not sure what to do with it.

    • Hercules AA has the same problem as Raptor.

- Reworked Teal AA laser

    • This version of Dagger has a bit of a whistle in its impact, and lacks weight on firing?

- Reworked Dagger

    • I don't get Halberd weapon impact. It sounds like a small building explosion rather than some sort of sci-fi magic missile.

-     • Created new halberd impact sound

    • Latest Lance is an improvement on [b1152ed](https://github.com/ZeroK-RTS/Zero-K/commit/b1152edf0425838c25eec8f5b806ce63c514d8be)
    • Buoy has the magic-energy firing sound without a matching impact, same as Halberd.

-     • Created new buoy sound

    • Widow is fine here.
    • Constable is very quiet when firing.

-     • Reworked slow beam

    • Moderator firing sound might be too short. It doesn't sound like it matches its impact.

-     • Extended and deepened moderator

    • Placeholder is fine here too.
    • This Jack is good.
    • Same Firewalker problem

-     • 100% reverted to base game

    • Toad doesn't sound like it deals damage.

- Reworked Teal AA laser

    • Ogre impact is offputting. The sound is very crackly.

-     • Redid ogre impact

    • I think this is a better Tremor firing sound, but I can't really hear the impacts.

-     • Mildly boosted the volume of impacts.  Tremor already has a big visual presence to inform players of it's impact. I want to be careful not to press this sound beyond informative into irritating.

    • Ettin sounds quite flat.

-     • re-created Ettin

    • The heatrays in Dante sound a bit unusual like Scorcher.

- Reworked Heatray for dante scorcher desolator

    • The latest Paladin is good.
    • This Detriment gauss is good. AA doesn't sound like it deals damage.

- Reworked Teal AA laser

I don't have a lot of time at the moment so I haven't been as detailed as I would like. The changes up to the present ([aac1dd7](https://github.com/ZeroK-RTS/Zero-K/commit/aac1dd77c90336d4e2854fb64f1b98f0193383b2)) are generally improvements. Here is what I would change:
    • Keep the previous Phantom sound. Or even just keep the firing sound. I tested soundStart = weapon/gauss_fire with the previous volume of the last commit and it works pretty well with the new impact sound. The new impact could be a bit brighter though, more dramatic.

-     • Made phanbtom impact more dramatic

    • The sustained windy part of the impact sound for Placeholder is quite loud, it should be quieter.

-     • 100% reverted placeholder to base game

    • Revert the Jack sound changes in [c911dc3](https://github.com/ZeroK-RTS/Zero-K/commit/c911dc39bc2ebf7333a5100b8898ec3e277dae1c)

-     • recreated jack

I've resolved a minor conflict related to other Zenith work.
Apologies again for not looking at the right versions. I think I can mostly summarise things as follows.
    • Some units sound like they lack damage. Bandit, Moderator, Constable, teal AA lasers, and heatrays.

-     • Adjusted each of these

    • Buoy and Halberd shoot energy projectiles that have the impact sound of mundane explosions.

- Adjusted these

    • Nimbus and Impaler should have their own weapon sounds, rather than taking them from weapons/units with distinct characteristics.

-     • Tried a new Nimbus.

    • A few impacts are a bit flat or offputting. Eg Ogre, Ripper, flak.

- 
Altered these

    • I don't know what to do about gauss and Firewalker.

- 
adjusted these

I am also not sure how I feel about sounds like Tremor and the new Zeno sound. They may end up working, but for now they sound a bit toy-like. I hope this is cleanness that will sound good when it all comes together.

- I think we are at the stage where having some more ears on this will be more informative.

GoogleFrog commented:
Thanks, a lot of things are improved and gauss seems to be basically there. I've gone through and created a list with the more minor issues as well. Many of them, especially the rocket launches and impacts, are likely preexisting.
- Rogue impact might be a bit too flat/low quality. It sounds less distinct compared to Thug impact.
- Thug impact is a bit too truncated. 
- Scythe still sounds like it has a slash with a more thumpy sound underneath it. Like dropping a large sheet of cardboard on the floor. Things that are a single action shouldn't sound like two actions as that creates event-noise. The new slash alone may work, but that said, the piercing nature of the previous slash is a feature, not a bug. Scythe needs to globally announce itself.
- Ronin firing might be a bit too flat.
- Sling impact is a bit too truncated, like Thug.
- Gremlin, Raptor etc still sound very underwhelming. Compare flying planes over a group of Gremlins to flying over a group of missile AA like Vandal or Angler. The plane dying doesn't square with the sound that is made is kill it. Part of it is that it sounds like a "recording studio" sound that isn't integrated into the world somehow. Perhaps a small amount of echo would fix it.
- Scorcher still sounds a bit weak and truncated. It sounds like individual shots rather than something that merges together into a ray. The new sound has a very noticeable point of maximum intensity that reinforces the barrage of shots feeling, and the feeling persists to multiple Scorchers. It takes 0.3s to extrude a ray, which the old sound was better at matching. I don't think a huge change would be needed though, perhaps making it slightly longer and softening the peak would work.
- Ripper impact is now very flat and echoy.
- Ravager impact truncated like Thug and Sling.
- Badger shot feels slightly too truncated. Shorter than the old one is good, but it could do with a bit of the rebound at the end that finishes it off.
- Impaler impact seems like it might have a good thump impact hiding underneath the sound of hitting a plastic box with raw pasta.
- Raptor see Gremlin.
- Magpie missile impact is a bit too truncated.
- The Blastwing death sound could be louder or more distinct, as it is barely audible. It is weird to see a "shot" fired without an accompanying sound. This is probably a preexisting issue.
- The Nimbus weapon sounds ok, although a bit generic and weak. A group of Nimbus firing is less loud, and probably less annoying, but I am not sure by how much. The impact sound generates a soft hum.
- Hercules see Gremlin.
- Dagger is much better. The impact sound is probably too loud, at least compared to the firing sound, but I would not increase the firing sound by very much, if at all.
- Scalpel impact is a bit too flat/low quality. This is mainly a problem when the missiles hit simultaneously, which is not that rare.
- Halberd impact still sounds like it is on the matter side of the matter/energy divide. But it is a lot better.
- Duck has the truncated Thug/Sling/Ravager sound, but I think it works here for a small missile. I can see a longer impact sound being a problem for spammy Duck, so perhaps the impact could be split off and so the plasma impacts could have a bit more tail.
- I like the new energy impact less for Buoy than I do for Halberd. 
- Bulkhead has the truncated impact too.
- Hermit has the truncated impact too.
- Puppy firing is a bit too flat. This makes sense as it is the same as the Ronin firing sound.
- Moderator is a lot better. I'd want to hear a version without the wet click about halfway through though, as removing it might improve it (or it might make the sound, idk).
- Jack weapon is quite good but is a touch too flat.
- Firewalker is Firewalker (for completeness)
- Toad has the Gremlin problem with sounding a bit weak/out of place. Also I can't hear the EMG.
- Ogre is a lot better. The impact sounds are too loud compared to firing sounds. I'd take the impact down more than I'd push the firing up.
- Tremor impacts probably need to be louder or slightly more noticeable in some way (higher pitch?). I know this is dangerous, but I can't relaly tell what I'm hearing when I see a rain of Tremor shots land somewhere. The firing sound also fails to be awesome, but in this case there is a tradeoff between singleplayer awe and multiplayer playability that is hard to satisfy, so the firing sound is probably fine.
- Ettin firing has a noticeable poweh sound that sounds like it clips over the rest of the sound, even though it isn't. This creates a clear repeat. The impact is very echoy and also a bit flat at the same time. I think it is an improvement though.
- Mistral impacts are a bit flat, like Rogue.
- Siren missile impact sounds weird for an impact. This is probably preexisting and not super important. The missile firing sound could be quieter if it wants.
- The truncated heatray is fine for Dante since so many other things are going on.
- Paladin rocket impact is too quiet compared to the rocket firing sound. The lightning could be a tad quieter if it needs to be.
- The Detriment gauss works well and also has space to be quieter. My comments about the volume don't really transfer across gauss reworks.
- I think the duration of the Faraday firing sound should be closer to the original, in a split from Venom. Faraday is rarely spammed, deals 2.5x the damage of Venom, and has a longer reload and EMP time. It sounds too underwhelming as is.
- Gauss impact sound could be softer, as with Dagger. Also I'm noticing that the gauss impact sound has a bit of a mid-pitch whistle at the end that would be good to remove.
- The lower fire rate of Thresher makes its flak sound a bit better.
- The Cerberus impact sound is quite underwhelming. The old dirt-scatter explosion was pretty bad, but then why does Shogun still use it?
- The Quake impact sound is too quiet for its rarity.
- The Shockley impact sound could be more noticeable somehow, which probably involves being less flat (like Imp). This sound should be noticeable as it tends to precede something important happening for most players in the game.
- I have come around to the Zeno sound distinction because it is nice to show that the weapon is not a slow wave (and thus deals friendly fire). 
- Starlight main firing could be louder.
- The DRP sound balance matches my missile silo comments. More impact/volume for green and blue.
- I rechecked the sea-only weapons and they're all good.
- Commander laser has a noticeable repeating shwss sound in it.

GoogleFrog commented:
Many of the minor things above are possibly due to the auditory environment, or what Zero-K "sounds like". The teal AA lasers are a good example. ZK sounds have some amount of echo and probably some other set of attributes that I don't know how to name. The nature of these sounds builds up a concept of how the environment of ZK passively modifies the "real" un-echoed sounds that we never hear. Eg the difference between being outside, in a bathroom, or an an-echoic chamber. The environmental behaviour encoded by the new teal AA laser is too far from the baseline established by the other sounds (and probably even by other games with a similar control scheme), so to me it feels detached from the game world.

GoogleFrog commented:
The "mundane" rocket and plasma cannons issues seem to be gone. Nimbus probably sounds as good as it can be, and still retain a decent amount of weight. Tremor firing works pretty well too. Halberd and Buoy seem to work. Here are some remaining things:

- I'm still not sure about the parts of the Scythe sound that aren't the clean slice noise. It sounds like distinct two distinct sounds to me and sounds weird on its own. When other stuff is going on, the extra whack just sort of disappears in more complicated situations. ZK has a level of stylisation that this sound seems to go out of its way to not match. But it's better than before so I'm happy to give it a try.
- I'm not sure why Phantom was changed but it seems fine.
- Gremlin is quite a bit better but probably a touch too flat. The sound is a lot less clear than other sounds. It works fairly well when a group of Gremlin are firing in semi-random ways. Raptor and Toad have trouble with their regular frequent fire.
- The new Ripper impact is far too weak. The original impact is what it should sound like, but higher quality/better balanced in some way.
- The Impaler is ok, but I'm not sure how it is better than the old one. Won't the maraca-type sound over the top of the deeper impact sound more annoying when spammed? There also seems to be extra echo at the end that echos the sound over the top, rather than the impact. This sound can happen a lot so shouldn't have too much going on. It's a bit like my Scythe comment, except that Impaler shots are much spammier and much less important than Scythe. So I'm not sure why it sort of sounds like two sounds put together.
- Dagger/Gauss have something weird going on. The start of the impact sound is good, but it turns into whistly flat sound at the end, that is almost like an attenuated echo. It gets worse when played multiple times one frame apart, which happens a lot given how gauss works. Also, if possible, it would probably be good to make the rising pitches in the tail of the firing sound quieter.
<img width="1669" height="769" alt="image" src="https://github.com/user-attachments/assets/77638348-8523-4e9e-bb39-004aa9a94f2d" />
- Not sure about Toad EMG, but will want to wait for the laser to work for other units.
- Cyclops firing sound could afford to be slightly longer. Was it truncated at some point?
- Tremor impact sound is a bit too loud. This will be hard to get right. Firing works well.
- Ettin impact sound is too echoy. It is a lot more echoy than its firing sound, which makes the two together sound weird.
- Envoy impact sound might be a bit too truncated, was it changed at some point?
- Shogun/Cerberus impacts are too clean and high pitched. They sound like smaller explosions than they are, and are a bit too far towards the energy weapon end of the spectrum. They should sound closer to Crab, Envoy and Emissary. Those are three quite distinct sounds, but I think Cerberus is further from all three in terms of weight and cleanliness.
- Detriment is still very loud. The impact sound is a lot better than the Dagger/Gauss one as it doesn't seem to be as flat. Hard to tell with all the Det noise though.
- The Starlight firing sound is in the `TARGETER` weapon.
- I think Blastwing can be left as-is. For future reference though, it has the weapon `TINY_BUILDINGEX` in `explodeAs` and `selfDestructAs`. This weapon can be found in the weapons directory, and has a sound.

GoogleFrog commented:
The list is getting quite short. The teal lasers seem good enough to test on people. I mainly don't understand what was wrong with the previous Ripper and Impaler.
- Ripper sounds very gravelly. I am not sure what it is supposed to be. The old boomph-pop fit together and was clear. I think ZK should generally sound a bit stylised/cartoony, and this Ripper impact is far from it.
- Impaler is better than Ripper, but to me it still sounds like a muddied version of the previous one. It sounds weaker and I don't know what the tradeoff is with the soundscape.
- Crasher firing has too much reverb.
- Dagger is pretty good. I'd go for more volume on the fire than the hit so it's easier to hear both events.
- Toad EMG basically works but could be less flat/echoy.
- Ettin and Thresher are still a bit weird. The firing and impact don't sound like they belong in the same environment. The firing sound is good but it might need to change to match everything else. Maybe add a bit of reverb for the firing sound? Not a lot, maybe something like the level of Ravager. 
- Making the Siren missile impact louder was probably a mistake now that I hear it. Maybe it could have a weaker version of whatever hit sound Impaler ends up with.
- Gauss fire and hit volume could be adjusted too. Gauss has longer range though, so the two sounds don't need the help that Dagger needs to be distinct.
- Cerberus and Shogun are still a bit echoy and lacking in weight. Eg they are higher pitched than Envoy which makes me think lower weight, plus they have more echo. This is a lot better though and much closer to where it should be. We probably can't have these weapons at their full weight given how spammy they are.

FrequentPilgrim commented:
"• Cerberus and Shogun are still a bit echoy and lacking in weight. Eg they are higher pitched than Envoy which makes me think lower weight, plus they have more echo. This is a lot better though and much closer to where it should be. We probably can't have these weapons at their full weight given how spammy they are."

- [ ] Made adjustments to pitch and echo. Not sure if it's better or just different, I may prefer the previous version but I want to get feedback and let it settle.

"• Toad EMG basically works but could be less flat/echoy."

- [ ] Reworked toad EMG into a new sound. Less shrill, less flat, more complimentary to the teal laser when toad fires both.

- [ ] Gave nimbus a little extra volume now that it can't run run away with volume when massing.

- [ ] I actually don't mind the lower fidelity of the ripper in that it sounds stylized and unque.

- [ ] While there's some potential extra stuff I could do, I don't want to pre-emptively start trying to solve theoretical problems that haven't revealed themselves to be actual problems. I'm feeling pretty good about where the latest round of changes has landed.
